src/components/PostComment.jsx

```diff
@@ -0,0 +1,79 @@
+import {useState,useEffect} from 'react'
+import { postCommentByArticleId } from '../../api'
+import styles from '../css/PostComment.module.css'
+
+function PostComment({article_id,username,comments,setComments}){
+
+    const [commentBody,setCommentBody] = useState('')
+    const [isFormValid,setIsFormValid] = useState(false)
+    const [isSubmittingBlankForm, setIsSubmittingBlankForm] = useState(false)
+
+   
+
+    useEffect(()=>{
+
+
+        if(isFormValid){
+            const today = new Date().toDateString()
+            const newResult = [...comments,{author:username,body:commentBody}]
+            setComments(newResult)
+
+            postCommentByArticleId(article_id,username,commentBody).then((comment)=>{     
```

Comment:
We don't want this reset to happen in a .then! At the moment, it's waiting for the post request to finish - we want to optimistically render the comment at the same time as making the request.